doc/hand_eye_calibration/hand_eye_calibration_tutorial.rst

```diff
@@ -0,0 +1,142 @@
+
+Launch RViz and Load Calibration Plugin
+---------------------------------------
+Launch the appropriate MoveIt demo for your robot. For instance, ``roslaunch panda_moveit_config demo.launch``.
+In the RViz "Panels" menu, choose "Add New Panel":
+
+.. image:: images/choose_new_panel.png
+
+Then, select the "HandEyeCalibration" panel type:
+
+.. image:: images/add_handeye_panel.png
+
+The panel will be added with the "Target" tab active.
+
+Create and Print a Target
+-------------------------
+Now we will create a visual calibration target. This target has distinctive patterns that are easy to identify in the
+image data, and by providing a measurement of the target size, the pose of the target in the camera's coordinate frame
+can be estimated. When conducting a hand-eye calibration, we do not need to know the target's precise location--as long
+as the target is stationary in the robot's base frame, the hand-eye calibration can be estimated from a sequence of 5 or
+more poses.
+
+In the "Target Params" section of the "Target" tab, we will use the default target parameters:
+
+- **markers, X**: 3
+- **markers, Y**: 4
+- **marker size (px)**: 200
+- **marker separation (px)**: 20
+- **marker border (bits)**: 1
+- **ArUco dictionary**: DICT_5X5_250
+
+Press the "Create Target" button to create the target image:
+
+.. image:: images/aruco_target_handeye_panel.png
+
+Save the target image using the "Save Target" button, and print out the image. Feel free to experiment with the target
+parameters to see how they affect the target, but be sure to remember the parameters used for the target you print--you
+will need to input the same parameters for the target to be recognized.
+
+The target must be flat to be reliably localized by the camera. Laying it on a flat surface is sufficient, or it can be
+mounted to a board. Measure the marker width (the outside dimension of one of the black squares), as well as the
+separation distance between markers. Enter these values, in meters, in the appropriate boxes in the "Target Params"
+section. Also, select the appropriate topics in the "Image Topic" and "CameraInfo Topic" drop-down menus.
+
+Finally, place the target near the robot, where it can be easily seen by the camera.
+
+Geometric Context
+-----------------
+The second tab, labeled "Context", contains the geometric information necessary to conduct the calibration. Set the
+"Sensor configuration" to "Eye-in-hand", then select the appropriate frames. The "Sensor frame" is the camera optical
+frame (using the right-down-forward standard, as specified in `REP 103 <https://www.ros.org/reps/rep-0103.html>`_), the
+"Object frame" is the frame defined by the calibration target, which is called "handeye_target" by default, and the
+remaining frames are the robot links rigidly attached to the camera (the "End-effector frame") and the calibration
+target (the "Robot base frame"; although the target might not be literally "rigidly attached", it doesn't move in this
```

Comment:
Nit: This paragraph could use a little tightening up / reformatting. This last sentence can be a bit confusing too, I think.

Reply:
I split it out into a numbered list.